spec/authorized_persona/authorization_spec.rb

```diff
@@ -330,13 +331,134 @@ def initialize(authorization_tier: nil)
     end
 
     it "is authorized for an arbitrary action if level action is set to :all" do
-      stub_const("User", user_class)
-      klass.authorize_persona(class_name: "User")
       klass.grant(three: :all)
 
       user = user_class.new(authorization_tier: "three")
 
       expect(klass.new(current_user: user, action: "bumpity")).to be_authorized
     end
+
+    context "with a specified current_user_method" do
+      let(:authorize_opts) { { class_name: "User", current_user_method: :custom_current_user } }
+
+      it "does not raise an error" do
+        user = user_class.new(authorization_tier: "four")
+        klass.define_method(:custom_current_user) { user }
+        klass.grant(four: 'show')
+
+        expect { klass.new(current_user: nil, action: "show").authorized? }.not_to raise_error
+      end
+    end
+
+    context 'with a non-symbol current_user method' do
+      let(:authorize_opts) { { class_name: "User" } }
+
+      it 'blows up' do
+        klass.grant(four: 'show')
+        klass.authorization_current_user_method = 'current_user'
+
+        expect { klass.new(current_user: nil, action: "show").authorized? }.to raise_error(
+          AuthorizedPersona::Error,
+          "you must configure authorization with a valid current_user method name, " \
+          "e.g. `authorize_persona class_name: 'User', current_user_method: :my_custom_current_user`",
+        )
+      end
+    end
+  end
+
+  describe '#authorize!' do
```

Comment:
This method was previously untested. I tried to unit test it, but it relies heavily on Rails controller APIs, so it may make sense to introduce true rails integration tests to this gem. (There are none right now.)

Reply:
i'm fine with testing like this for this PR and then exploring integration tests. just to say here what i voiced over in-person: i don't love relying on stubbing out all these internal Rails APIs, even if they are relatively stable, but i don't actually have anything good to offer about why not other than the tests are arguably made more brittle and harder to understand when they have to stub rails internal machinations.

Reply:
...and we start getting into territory where if rails changes these tests might continue to pass even if the behavior doesn't actually work anymore.